.github/workflows/main.yml

```diff
@@ -174,9 +181,14 @@ jobs:
         with:
           deno-version: "1.18.2"
 
-      - name: get binary and make executable from previous step for tests
+      - name: build binary and make executable
+        env:
+          TAQ_VERSION: ${{ github.ref_name }}
+          DENO_DIR: "./deno"
         run: |
-          wget ${{ needs.build-binaries.outputs.public-url-ubuntu }} -O taq
+          COMMIT=`git rev-parse --short "$GITHUB_SHA"`
+          deno compile --output taq --allow-run --allow-write --allow-read --allow-env index.ts --quickstart "`cat quickstart.md`" --setBuild "$TAQ_VERSION/$COMMIT" --setVersion "$TAQ_VERSION"
```

Comment:
I think it's worthwhile to split this out and make this local compile only happen when coming from a forked repo and download the file from GCP otherwise. That way for internal PRs we still download the binary for specific build targets and forked PRs will have this generic binary built for them.

I mostly want to leave the door open for my other PR that will have the tests run on each binary that we produce (linux and mac)

Reply:
Sure thing. I added a another step and a condition that would use one or the other based on the PR coming from a fork or not. Because we would need the `build-binaries` job to complete first even if it is skipped, I had to add another two conditions to the `run-tests` job.

Reply:
Awesome, thank you so much Gino